src/Parser.php

```diff
@@ -26,13 +26,15 @@
  */
 class Parser
 {
-    const INVALID_URI_CHARS = "\x00\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0A\x0B\x0C\x0D\x0E\x0F\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1A\x1B\x1C\x1D\x1E\x1F\x7F";
+    const INVALID_CHARS_REGEXP = '/[\x00-\x1f\x7f]/';
 
-    const SCHEME_VALID_STARTING_CHARS = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+    const SCHEME_REGEXP = '/^[a-z][a-z\+\.\-]*$/i';
 
-    const SCHEME_VALID_CHARS = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789+.-';
+    const HOST_REGEXP = '/^(?:[a-z0-9\-_]{1,63}\.){0,126}[a-z0-9\-_]{1,63}$/i';
 
-    const LABEL_VALID_STARTING_CHARS = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+    const DOMAIN_LABEL_REGEXP = '/^[a-z0-9\-_]{1,63}$/i';
+
+    const LABEL_REGEXP = '/^[a-z0-9!$&\-\'\(\)*+,;=_~]+$/i';
```

Comment:
This should be `HOST_LABEL_REGEXP` to avoid confusion with `DOMAIN_LABEL_REGEXP`.